Xamarin.Forms.Core/Shell/BaseShellItem.cs

```diff
@@ -246,6 +251,166 @@ EffectiveFlowDirection IFlowDirectionController.EffectiveFlowDirection
 		internal virtual void ApplyQueryAttributes(IDictionary<string, string> query)
 		{
 		}
+
+
+		internal override void OnStyleClassChanged()
+		{
+			if (_defaultFlyoutItemCell == null)
+				return;
+
+			base.OnStyleClassChanged();
+			List<string> bindableObjectStyle = new List<string>() {
+				DefaultFlyoutItemLabelStyle,
+				DefaultFlyoutItemImageStyle,
+				DefaultFlyoutItemStyle };
+
+			if (StyleClass != null)
+				foreach (var styleClass in StyleClass)
+					bindableObjectStyle.Add(styleClass);
+
+			_defaultFlyoutItemCell.StyleClass = bindableObjectStyle;
+			_defaultFlyoutItemCell.Children.OfType<Label>().First()
+				.StyleClass = bindableObjectStyle;
+			_defaultFlyoutItemCell.Children.OfType<Image>().First()
+				.StyleClass = bindableObjectStyle;
+		}
+
+		internal DataTemplate CreateDefaultFlyoutItemCell(string textBinding, string iconBinding)
+		{
+			return new DataTemplate(() =>
+			{
+				if (_defaultFlyoutItemCell != null)
+					return _defaultFlyoutItemCell;
+
+				var grid = _defaultFlyoutItemCell = new Grid();
+				if (Device.RuntimePlatform == Device.UWP)
```

Comment:
ew

Reply:
So this is mainly just pulling up what was already defined on ios/android

It just centralizes this template
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/app-fundamentals/shell/flyout#default-template-for-flyoutitems-and-menuitems

Into one place on the core code so that the styles can be applied from the cross platform layer and to reduce duplication of the template itself

But let me mess with it a bit.... Since we're using styles now I might be able to just set the xplat classes to one thing and then apply the platform level styles at the platform level

Reply:
I feel like this is fine because it's the xaml control template representing the flyoutitems.

Once/if we start building composite controls this is probably what the templates are going to look like. They'll have OnPlatforms trickled throughout representing things that were configured differently.